sklearn/neighbors/approximate.py

```diff
@@ -309,15 +318,23 @@ def _get_radius_neighbors(self, query, max_depth, bin_queries, radius):
         total_neighbors = np.array([], dtype=int)
         total_distances = np.array([], dtype=float)
 
+        index_offsets = bin_queries // (2**self._residual)
+
         while (max_depth > self.min_hash_match and
                ratio_within_radius > threshold):
             left_mask = self._left_mask[max_depth]
             right_mask = self._right_mask[max_depth]
             candidates = []
             for i in range(self.n_estimators):
-                start, stop = _find_matching_indices(self.trees_[i],
+                start, stop = _find_matching_indices(self.trees_[i]
+                                                     [self._locations
+                                                      [i, index_offsets[i]]:
```

Comment:
Having this moved into `_find_matching_indices` affects `_find_longest_prefix_match` as well. My idea is that there is no point of having both `min_hash_match` and `offset`(or whatever we call it). What we need to do is using the one both purposes and getting rid of the other. WDYT?